### PR TITLE
Save and handle post login redirect

### PR DESCRIPTION
**Pull request by kate-shine**
squashed commits from https://github.com/ctron/yew-oauth2/pull/25 and rebased on main

obsoletes https://github.com/ctron/yew-oauth2/pull/25 https://github.com/ctron/yew-oauth2/pull/28

### CODE REVIEW

**Comment by kate-shine** [comment]
That was a really weird rebase O.o Worked well after squashing, but before that, it was really messy for some reason

File: Cargo.toml [hunk 1]
```diff
@@ -33,7 +33,7 @@ web-sys = { version = "0.3", features = [
 ] }
 
 openidconnect = { version = "3.0", optional = true }
-yew-nested-router = { version = ">=0.5, <0.7", optional = true }
+yew-nested-router = { version = ">=0.6.3, <0.7", optional = true }
```
Comment (ctron) [inline Cargo.toml:36]: I guess it's time to bring this back to just `0.6.3`. But I can do this post-merge.

**Review by ctron — APPROVED** [review]
This is awesome, and I learned something new. Thanks for the PR!

File: src/agent/mod.rs [hunk 2]
```diff
@@ -362,6 +393,24 @@ where
         }
     }
 
+    fn post_login_redirect(&self) -> Result<(), OAuth2Error> {
+        let config = self.config.as_ref().ok_or(OAuth2Error::NotInitialized)?;
+        let Some(redirect_callback) = config
```
[the remark this answers is not in the document]
Comment (kate-shine) [reply src/agent/mod.rs:398]: It's new-ish and super convenient for return early pattern :)

**Pull request merged**